modules/javafx.graphics/src/main/java/javafx/scene/layout/Border.java

```diff
@@ -300,83 +306,85 @@ public Border(@NamedArg("strokes") List<BorderStroke> strokes, @NamedArg("images
+     * and preserve the immutability invariant.
+     *
+     * @param strokes the strokes, not {@code null}
+     * @param images the images, not {@code null}
+     */
+    private Border(List<BorderStroke> strokes, List<BorderImage> images, int ignored) {
+        Objects.requireNonNull(strokes, "strokes cannot be null");
+        Objects.requireNonNull(images, "images cannot be null");
+
         double innerTop = 0, innerRight = 0, innerBottom = 0, innerLeft = 0;
         double outerTop = 0, outerRight = 0, outerBottom = 0, outerLeft = 0;
 
-        if (strokes == null || strokes.length == 0) {
-            this.strokes = Collections.emptyList();
-        } else {
-            final BorderStroke[] noNulls = new BorderStroke[strokes.length];
-            int size = 0;
-            for (int i=0; i<strokes.length; i++) {
-                final BorderStroke stroke = strokes[i];
-                if (stroke != null) {
-                    noNulls[size++] = stroke;
-
-                    // Calculate the insets and outsets. "insets" are the distance
-                    // from the edge of the region to the inmost edge of the inmost border.
-                    // Outsets are the distance from the edge of the region out towards the
-                    // outer-most edge of the outer-most border.
-                    final double strokeInnerTop = stroke.innerEdge.getTop();
-                    final double strokeInnerRight = stroke.innerEdge.getRight();
-                    final double strokeInnerBottom = stroke.innerEdge.getBottom();
-                    final double strokeInnerLeft = stroke.innerEdge.getLeft();
-
-                    innerTop = innerTop >= strokeInnerTop ? innerTop : strokeInnerTop;
-                    innerRight = innerRight >= strokeInnerRight? innerRight : strokeInnerRight;
-                    innerBottom = innerBottom >= strokeInnerBottom ? innerBottom : strokeInnerBottom;
-                    innerLeft = innerLeft >= strokeInnerLeft ? innerLeft : strokeInnerLeft;
-
-                    final double strokeOuterTop = stroke.outerEdge.getTop();
-                    final double strokeOuterRight = stroke.outerEdge.getRight();
-                    final double strokeOuterBottom = stroke.outerEdge.getBottom();
-                    final double strokeOuterLeft = stroke.outerEdge.getLeft();
-
-                    outerTop = outerTop >= strokeOuterTop ? outerTop : strokeOuterTop;
-                    outerRight = outerRight >= strokeOuterRight? outerRight : strokeOuterRight;
-                    outerBottom = outerBottom >= strokeOuterBottom ? outerBottom : strokeOuterBottom;
-                    outerLeft = outerLeft >= strokeOuterLeft ? outerLeft : strokeOuterLeft;
-                }
-            }
-            this.strokes = new UnmodifiableArrayList<>(noNulls, size);
+        for (int i = 0, max = strokes.size(); i < max; i++) {
+            final BorderStroke stroke = strokes.get(i);
```

Comment:
just curious: do we really need `final` keyword for local variables after java8?

Reply:
I don't use it, but I didn't change the existing code here.

Reply:
> just curious: do we really need `final` keyword for local variables after java8?

It's superfluous, and only useful if you are not initializing it immediately and you still want to force a single assignment.  If you always immediately initialize locals (either using a method call or through a ternary (`?:`)) then the `final` keyword is definitely just noise.